src/qir2qasm_trans/qir_trans/translator.py

```diff
@@ -0,0 +1,343 @@
+                    func_builder = DefCalBuilder(func_name)
+                    func_info = FunctionInfo(
+                        type=func_type.as_ir(self.profile.context),
+                        def_statement=None,
+                        builder=func_builder,
+                    )
+                    self.symbols.functions[func.name] = (
+                        func_info  # Register Function in SymbolTable
+                    )
+
+                    statements_declaration.append(
+                        ast.CalibrationDefinition(
+                            name=ident,
+                            arguments=arguments,
+                            qubits=qubits,
+                            return_type=ret_type_ast,
+                            body="",
+                        )
+                    )
+
+        return statements_declaration + statements_definition
+
+    def build_main(self) -> List[ast.Statement]:
+        main_func = None
+        for func in self.module.functions:
+            # check main function definition
+            attr_dict = {}
+            for attr in func.attributes:
+                matches = re.findall(r'"(\w+)"(?:="([^"]*)")?', attr.decode())
+                for k, v in matches:
+                    if v != "":
+                        attr_dict[k] = v
+                    else:
+                        attr_dict[k] = True
+
+            if "entry_point" in attr_dict.keys():
+                main_func = func
+                break
+        else:
+            raise Exception("No main function defined!")
+
+        statements = []
+        self.entry_block = main_func.name
+        for block in main_func.blocks:
+            # self.build_block(block)
+            statements.extend(self.build_block(block))
+
+        self.build_control()
+        return statements
+
+    def build_block(self, block: ValueRef) -> List[ast.Statement]:
+        statements = []
+        block_name = block.name
+        for inst in block.instructions:
+            statements.extend(self.build_instruction(inst, block_name))
+        self.symbols.block_statements[block_name] = statements
+        return statements
+
+    def build_control(self):
+        is_updated = True
+        while is_updated:
+            self.create_CFG()
+            is_updated = False
+
+    def create_CFG(self) -> nx.DiGraph:
+        cfg = nx.DiGraph()
+        for block, block_info in self.symbols.block_branchs.items():
+            for tgt_block in block_info.br_tgt:
+                cfg.add_edge(block, tgt_block)
+        return cfg
+
+    def build_instruction(self, instruction: ValueRef, block_name: str):
+        statements = []
+        if instruction.opcode in _LLVM_INSTRUCTIONS.keys():
+            statements.extend(self.build_llvm_inst(instruction))
+        elif instruction.opcode == "call":
+            statements.extend(self.build_func_call(instruction))
+        elif instruction.opcode == "br":
```

Comment:
Is `br` defined by LLVM? If not, we would prefer a more explicit naming like "branch" instead of "br"